src/metrics/process.rs

```diff
@@ -0,0 +1,210 @@
+use crate::metrics::config::Config;
+// use aws_lambda_events::encodings::Base64Data;
+use aws_lambda_events::event::firehose::{KinesisFirehoseResponse, KinesisFirehoseEvent, KinesisFirehoseResponseRecord, KinesisFirehoseResponseRecordMetadata};
+use lambda_runtime::Error;
+use opentelemetry_proto::tonic::collector::metrics::v1::ExportMetricsServiceRequest;
+use opentelemetry_proto::tonic::common::v1::any_value;
+use opentelemetry_proto::tonic::common::v1::AnyValue;
+use opentelemetry_proto::tonic::common::v1::KeyValue;
+use prost::encoding::{decode_varint, encode_varint};
+use prost::Message;
+use reqwest;
+use tracing::{debug, error};
+
+
+fn split_length_delimited(data: &[u8]) -> Result<Vec<&[u8]>, String> {
+    let mut chunks = Vec::new();
+    let mut remaining = data;
+
+    while remaining.len() > 0 {
+        // Decode the length prefix (varint)
+        let length = decode_varint(&mut remaining)
+            .map_err(|e| format!("Failed to decode varint: {}", e))? as usize;
+
+        // Ensure the remaining data is enough for the length
+        if length > remaining.len() {
+            return Err("Insufficient data for length-delimited field".to_string());
+        }
+
+        // Extract the length-delimited chunk
+        let chunk = &remaining[0..length];
+        chunks.push(chunk);
+
+        // Move the position forward
+        remaining = &remaining[length..];
+    }
+
+    Ok(chunks)
+}
+
+// TODO: look into whether these messages can be batched instead of sending one at a time
+#[allow(dead_code)]
```

Comment:
Did you intend to keep this code?

Reply:
This function was written as an attempt to re-batch metric data after transforming the individual records. Instead of sending messages one  by one, the goal was to send a batch as Firehose originally does. This however, did not work. 

Leaving the function in so we can revisit this in the future.